pkg/apis/eventing/v1alpha1/trigger_lifecycle.go

```diff
@@ -63,14 +73,22 @@ func (ts *TriggerStatus) PropagateBrokerStatus(bs *BrokerStatus) {
 		if bc := brokerCondSet.Manage(bs).GetCondition(BrokerConditionReady); bc != nil {
 			msg = bc.Message
 		}
-		ts.MarkBrokerFailed("BrokerNotReady", "Broker is not ready: %s", msg)
+		if bs.IsUnknown() {
+			ts.MarkBrokerUnknown("BrokerUnknown", "The status of Broker is Unknown: %s", msg)
+		} else {
```

Comment:
this is incorrect. bs could be `true`.

Reply:
We already checked scenario `True` in above logic